### DISCUSSION TITLE
How to get the battery level of a connected device ?

### DISCUSSION THREAD

**Discussion by pythops** [bluez/bluer#123] [Q&A]
How to get the battery level of a connected device ?

**Comment by surban** [comment]
The battery API is currently not implemented in bluer, but a PR would be welcome.

**Comment by pythops** [comment]
Absolutely ! 
I'm new to the bluetooth world, so if you have some guidelines on how to implement it that would be very great :pray:

**Reply by surban** [reply]
You will need to add a property to [`Device`](https://github.com/bluez/bluer/blob/aa9477c01aca72cc5aa4a583db603c0857677d39/bluer/src/device.rs#L274) that queries the battery status using the [battery D-Bus interface](https://github.com/bluez/bluez/blob/master/doc/org.bluez.Battery.rst). This will look like the other properties, but with the interface `org.bluez.Battery1`.

**Reply by pythops** [reply]
thanks for your guidelines :pray:
